experiment/popularity_community/crawl_torrents.py

```diff
@@ -0,0 +1,244 @@
+        unpacked_data = self.mds.process_squashed_mdblob(decompressed_data)
+        # ↑ above some magic from the base class
+
+        for metadata, _ in unpacked_data:
+            if not metadata:
+                continue
+
+            self.add_torrent_to_the_list(peer, metadata)
+
+    def add_torrent_to_the_list(self, peer, metadata):
+        key = hexlify(peer.mid)
+        self._logger.debug(f"Decompress torrent item for: {key}")
+
+        votes = int(getattr(metadata, 'votes', 0))
+        infohash = hexlify(getattr(metadata, 'infohash', ''))
+        title = getattr(metadata, 'title', '')
+
+        self.torrent_list[key].append((votes, infohash, title))
+
+    def save(self):
+        self._logger.debug(f"Save results to : {self._configuration.output_file_name}")
+
+        with open(self._configuration.output_file_name, 'w') as json_file:
+            json.dump(self.torrent_list, json_file)
+
+    def introduction_response_callback(self, peer, dist, payload):
+        """ This method adds logging behaviour to the base implementation
+        """
+        super().introduction_response_callback(peer, dist, payload)
+        self._logger.debug(f"Peer {peer} wants to introduce {payload.wan_introduction_address}")
+
+
+class Service(TinyTriblerService):
+    def __init__(self, configuration, working_dir, config_path):
+        super().__init__(Service.create_config(working_dir, config_path),
+                         configuration.timeout_in_sec,
+                         working_dir,
+                         config_path)
+        self._configuration = configuration
+
+    @staticmethod
+    def create_config(working_dir, config_path):
+        config = TinyTriblerService.create_default_config(working_dir, config_path)
+
+        config.set_ipv8_enabled(True)
+        config.set_ipv8_walk_interval = ExperimentConfiguration.IPV8_WALK_INTERVAL
+
+        config.set_chant_enabled(True)
+        return config
+
+    async def on_tribler_started(self):
+        await super().on_tribler_started()
+        session = self.session
+        peer = Peer(session.trustchain_keypair)
+
+        session.remote_query_community_community = Crawler(peer, session.ipv8.endpoint,
+                                                           session.ipv8.network,
+                                                           session.mds, self._configuration)
+
+        session.ipv8.overlays.append(session.remote_query_community_community)
+        session.ipv8.strategies.append((RandomWalk(session.remote_query_community_community,
+                                                   timeout=ExperimentConfiguration.RANDOM_WALK_TIMEOUT_IN_SEC,
+                                                   window_size=ExperimentConfiguration.RANDOM_WALK_WINDOW_SIZE,
+                                                   reset_chance=ExperimentConfiguration.RANDOM_WALK_RESET_CHANCE),
+                                        UNLIMITED))
+
+
+def _parse_argv(argv):
```

Comment:
You can use the `argparse` library to simplify the parsing of arguments, for example, see [here](https://github.com/Tribler/application-tester/blob/master/main.py#L10).

Reply:
👍 thanks!

Reply:
Code has been improved.